core/src/main/scala/org/apache/spark/sql/extensions/rules.scala

```diff
@@ -26,7 +26,8 @@ class TiAuthRuleFactory(getOrCreateTiContext: SparkSession => TiContext)
     extends (SparkSession => Rule[LogicalPlan]) {
   private val logger = LoggerFactory.getLogger(getClass.getName)
   override def apply(sparkSession: SparkSession): Rule[LogicalPlan] = {
-    if (TiExtensions.authEnable(sparkSession)) {
+    // auth only work for catalog plugin mode
+    if (TiExtensions.catalogPluginMode(sparkSession) && TiExtensions.authEnable(sparkSession)) {
```

Comment:
We should check catalogPluginMode before all factories, and throw Exception if catalogPluginMode is off.
SInce nonecatalogPluginMode is not supported anymore, we don't need to maintain else condition in all factories

Reply:
i will try to do it